### PR TITLE
feature(xcy): add ReZero algo. and related configs

### PR DESCRIPTION
**Pull request by puyuan1996**
- Our arxiv paper is on [ReZero: Boosting MCTS-based Algorithms by Backward-view and Entire-buffer Reanalyze](https://arxiv.org/abs/2404.16364).

### CODE REVIEW

File: lzero/entry/utils.py [hunk 1]
```diff
@@ -75,3 +76,45 @@ def log_buffer_memory_usage(train_iter: int, buffer: "GameBuffer", writer: Summa
...
+
+        # Calculate and log average reuse search time.
+        average_reuse_search_time = buffer.reuse_search_time / sample_times
+        writer.add_scalar('Buffer/average_reuse_search_time', average_reuse_search_time, train_iter)
+
+        # Calculate and log average active root number.
+        average_active_root_num = buffer.active_root_num / sample_times
+        writer.add_scalar('Buffer/average_active_root_num', average_active_root_num, train_iter)
+
+        # Reset the time records in the buffer.
```
Comment (PaParaZz1) [inline lzero/entry/utils.py:115]: this reset method should be set in buffer and called by outside

File: lzero/entry/utils.py [hunk 2]
```diff
@@ -75,3 +76,45 @@ def log_buffer_memory_usage(train_iter: int, buffer: "GameBuffer", writer: Summa
...
+def log_buffer_run_time(train_iter: int, buffer: "GameBuffer", writer: SummaryWriter) -> None:
+    """
+    Overview:
+        Log the average runtime metrics of the buffer to TensorBoard.
+    Arguments:
+        - train_iter (:obj:`int`): The current training iteration.
+        - buffer (:obj:`GameBuffer`): The game buffer containing runtime metrics.
+        - writer (:obj:`SummaryWriter`): The TensorBoard writer for logging metrics.
+
+    Note:
```
Comment (PaParaZz1) [inline lzero/entry/utils.py:90]: .. note::

File: lzero/mcts/buffer/game_segment.py [hunk 3]
```diff
@@ -210,7 +210,10 @@ def store_search_stats(
             store the visit count distributions and value of the root node after MCTS.
         """
         sum_visits = sum(visit_counts)
-        if idx is None:
+        if sum_visits == 0:
```
Comment (PaParaZz1) [inline lzero/mcts/buffer/game_segment.py:213]: add comments for this branch

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: zoo/atari/config/atari_rezero_ez_config.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            decay=int(1e5),
+        ),
+        use_augmentation=True,
+        update_per_collect=update_per_collect,
+        batch_size=batch_size,
+        optim_type='SGD',
+        lr_piecewise_constant_decay=True,
+        learning_rate=0.2,
+        num_simulations=num_simulations,
+        reanalyze_ratio=0,  # NOTE: for rezero, reanalyze_ratio should be 0. To be compatible with muzero.
```
[the remark this answers is not in the document]
Comment (puyuan1996) [reply zoo/atari/config/atari_rezero_ez_config.py]: ReZero是通过buffer_reanalyze_freq控制reanalyze频率的，这个参数用不到，但为了与muzero兼容，保留它为默认值。

**Pull request merged**